### PR TITLE
Client with flag for memory tracker

### PR DESCRIPTION
**Pull request by FArthur-cmd**
I hereby agree to the terms of the CLA available at: https://yandex.ru/legal/cla/?lang=en

Changelog category (leave one):
- Improvement

Changelog entry (a user-readable short description of the changes that goes to CHANGELOG.md):
Memory client in client

Detailed description / Documentation draft:
It is other version of #26669, because it was easier to write it from begging, than to revert changes.

# Memory tracker

You can use flag `--max_memory_usage_in_client` to set max memory usage (by default it is set to zero, which means unlimited).

### CODE REVIEW

**Review by kitaisreal — APPROVED** [review]

**Comment by kitaisreal** [comment]
@FArthur-cmd please add some tests.

**Comment by FArthur-cmd** [comment]
I found out that it doesn't work correctly. For example, if you set `max_memory_usage_for_client=1`. It will have exception in suggest, but later it will continue working like there is no memory limit.

**Comment by kitaisreal** [comment]
@FArthur-cmd fix big reference in test.

**Comment by FArthur-cmd** [comment]
@kitaisreal This big reference helps to show cases where and how memory_tracker works. 
1) more then untracked memory
2) less then untracked memory
3) without explicit tracker limit everything works

**Comment by kitaisreal** [comment]
@FArthur-cmd we cannot add reference for 60k lines, consider somehow to change its size or write some sort of integration test.

**Comment by FArthur-cmd** [comment]
@Mergifyio refresh

File: programs/client/Client.cpp
Comment (kitaisreal) [inline programs/client/Client.cpp:543]: @FArthur-cmd  consider to use config().getInt64("max_memory_usage_in_client", 0 /*default value*/);

File: programs/client/Client.cpp
Comment (kitaisreal) [inline programs/client/Client.cpp:319]: Fix formatting.

**Review by kitaisreal — APPROVED** [review]

**Review by kitaisreal — APPROVED** [review]

File: tests/queries/0_stateless/02003_memory_limit_in_client.expect
Comment (azat) [inline tests/queries/0_stateless/02003_memory_limit_in_client.expect:1]: @FArthur-cmd it seems that this test can be an `sh` test, I converted it here - https://github.com/ClickHouse/ClickHouse/pull/51475

But maybe I missed something? And interactivity is indeed necessary?

Comment (FArthur-cmd) [reply tests/queries/0_stateless/02003_memory_limit_in_client.expect:1]: As far as I remember, this was done to check the expected behavior when running the client. It was necessary to check that the memory limits worked correctly and would not break the client.

Comment (azat) [reply tests/queries/0_stateless/02003_memory_limit_in_client.expect:1]: Ok, then we can convert this to something like:

```
$CLICKHOUSE_CLIENT --max_memory_usage_in_client=1 -n -q "
SELECT arrayMap(x -> range(x), range(number)) FROM numbers(1000); -- { clientError MEMORY_LIMIT_EXCEEDED }
SELECT * FROM (SELECT * FROM system.numbers LIMIT 600000) as num WHERE num.number=60000;
"
```

This should be the same.

But to do this, first need to wait until #51494 will be merged

**Pull request merged**
